src/backends/tensorrt/plan_backend.cc

```diff
@@ -611,7 +645,8 @@ PlanBackend::CreateExecutionContext(
   RETURN_IF_ERROR(
       context->InitEventSet(Config().optimization().cuda().busy_wait_events()));
 
-  auto eit = device_engines_.find(gpu_device);
+  auto device_pair = std::make_pair(gpu_device, dla_core_id);
+  auto eit = device_engines_.find(device_pair);
   if (eit->second.second == nullptr) {
```

Comment:
If seems like the runtime should never be null and that is_shared_engine_ and related code can be removed. @tanmayv25  do you agree?

Reply:
Right. For the new case models that have same GPU and no DLA enabled are represented with <gpu_id, -1> and with dla enabled would be <gpu_id, dla_core_id>
Sharing of engines for cases that have same same gpu_id and dla_core_id (where -1 means no dla core is used) would only happen for models that have no dynamic shapes and are not sequence models.